### ISSUE TITLE
Add default support for YAML as a default template/directory data file format

### ISSUE THREAD

**Issue by zachleat** [11ty/eleventy#3228]
* `posts/subdir/my-first-blog-post.11tydata.js`
* `posts/subdir/my-first-blog-post.11tydata.json`
* Proposed addition: `posts/subdir/my-first-blog-post.11tydata.yaml`
* `posts/subdir/my-first-blog-post.json`

Preference for `yaml` via https://yaml.org/faq.html

Related docs: https://www.11ty.dev/docs/data-template-dir/

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by cj81499** [comment]
Although I personally prefer yaml over JSON for most situations, it's trivial to [add it as a custom data format](https://www.11ty.dev/docs/data-custom/#yaml) and this is a pretty minor preferencs to the point that I typically use json due to it's ubiquity in my projects anyways.

Additionally, it seems best to avoid needlessly increasing 11ty's bundle/download size by adding a dependency on a yaml parser that many users won't use.

In short, I don't see significant value in supporting yaml out of the box, but if 11ty _already_ requires a yaml parser (I suspect it does for front matter), I see no harm in supporting yaml out of the box.

**Comment by pepelsbey** [comment]
I use YAML data files in all my projects via the mentioned custom data format. Although it’s trivial to add, I think it should be supported by default. One of the use cases for me looks like this:

```
post/
  index.yml
  index.md
```

Instead of using front matter in markdown files, I create index.yml file and store all the data I need there. It also allows me to use tabs in my project, with exception to YAML files via EditorConfig, since tabs are not allowed in YAML for some reason.

**Comment by kleinfreund** [comment]
In theory, it should be possible to rely on a YAML parser alone for parsing both YANL and JSON because YAML is a superset of JSON and so any valid JSON should be able to be parsed by a YAML parser. That would mitigate bundle size increases. 

Other than that I don’t think YAML necessarily needs to be provided by default since adding it via configuration is easy.

**Comment by cj81499** [comment]
@kleinfreund 

> That would mitigate bundle size increases.

How would this mitigate bundle size? JSON parsing is built into the JavaScript standard library, so it has effectively no bundle size impact.
